packages/nouns-contracts/contracts/governance/ExcessETHBurner.sol

```diff
@@ -0,0 +1,247 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/**
+ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *
+ */
+
+pragma solidity ^0.8.19;
+
+import { IExcessETHBurner } from '../interfaces/IExcessETHBurner.sol';
+import { Ownable } from '@openzeppelin/contracts/access/Ownable.sol';
+import { IERC20 } from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import { INounsAuctionHouseV2 } from '../interfaces/INounsAuctionHouseV2.sol';
+
+interface RocketETH {
+    function getEthValue(uint256 _rethAmount) external view returns (uint256);
+}
+
+interface INounsDAOV3 {
+    function adjustedTotalSupply() external view returns (uint256);
+}
+
+interface IExecutorV3 {
+    function burnExcessETH(uint256 amount) external;
+}
+
+/**
+ * @title ExcessETH Burner
+ * @notice A helpder contract for burning Nouns excess ETH with NounsDAOExecutorV3.
+ * @dev Owner is assumed to be the NounsDAOExecutorV3 contract, i.e. the Nouns treasury.
+ */
+contract ExcessETHBurner is IExcessETHBurner, Ownable {
+    /**
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     *   ERRORS
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     */
+
+    error NotTimeToBurnYet();
+    error NoExcessToBurn();
+    error NotEnoughAuctionHistory();
+    error PastAuctionCountTooLow();
+
+    /**
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     *   EVENTS
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     */
+
+    event AuctionSet(address oldAuction, address newAuction);
+    event NextBurnNounIDSet(uint128 nextBurnNounID, uint128 newNextBurnNounID);
+    event MinNewNounsBetweenBurnsSet(uint128 minNewNounsBetweenBurns, uint128 newMinNewNounsBetweenBurns);
+    event NumberOfPastAuctionsForMeanPriceSet(
+        uint16 oldNumberOfPastAuctionsForMeanPrice,
+        uint16 newNumberOfPastAuctionsForMeanPrice
+    );
+
+    /**
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     *   STATE
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     */
+
+    uint256 public constant MIN_PAST_AUCTIONS = 2;
+
+    INounsDAOV3 public immutable dao;
+    INounsAuctionHouseV2 public immutable auction;
+    IERC20 public immutable wETH;
+    IERC20 public immutable stETH;
+    IERC20 public immutable rETH;
+    uint128 public nextBurnNounID;
+    uint128 public minNewNounsBetweenBurns;
+    uint16 public numberOfPastAuctionsForMeanPrice;
+
+    constructor(
+        address owner_,
```

Comment:
nit: let's make sure to add natspec here that states that `owner_` is expected to be a timelock, implementing `IExecutorV3`